generator/utils.go

```diff
@@ -0,0 +1,282 @@
+
+	langRE := regexp.MustCompile(`\s*@lang (.+)`)
+	paramTitleRE := regexp.MustCompile(`\s*@parameter title (\w+): (.+)`)
+	paramDescRE := regexp.MustCompile(`\s*@parameter description (\w+): (.+)`)
+	balanceRE := regexp.MustCompile(`\s*@balance (\w+): (.+)`)
+
+	langMatch := langRE.FindStringSubmatch(codeCommentBlock)
+	if langMatch == nil {
+		langMatch = []string{"", "en-US"}
+	}
+	messageTitleMatch := messageTitleRE.FindStringSubmatch(codeCommentBlock)
+	if len(messageTitleMatch) > 0 {
+		// Populate the template with extracted data
+		if template.Data.Messages.Title == nil {
+			template.Data.Messages.Title = &flixkit.Title{
+				I18N: map[string]string{
+					"en-US": "",
+				},
+			}
+		}
+		template.Data.Messages.Title = &flixkit.Title{
+			I18N: map[string]string{
+				langMatch[1]: messageTitleMatch[1],
+			},
+		}
+	}
+
+	messageDescMatch := messageDescRE.FindStringSubmatch(codeCommentBlock)
+	if len(messageDescMatch) > 0 {
+		if template.Data.Messages.Description == nil {
+			template.Data.Messages.Description = &flixkit.Description{
+				I18N: map[string]string{
+					"en-US": "",
+				},
+			}
+		}
+		template.Data.Messages.Description = &flixkit.Description{
+			I18N: map[string]string{
+				langMatch[1]: messageDescMatch[1],
+			},
+		}
+	}
+
+	paramTitleMatches := paramTitleRE.FindAllStringSubmatch(codeCommentBlock, -1)
+	paramDescMatches := paramDescRE.FindAllStringSubmatch(codeCommentBlock, -1)
+	balanceMatches := balanceRE.FindAllStringSubmatch(codeCommentBlock, -1)
+
+	for _, match := range paramTitleMatches {
+		argName := match[1]
+		argTitle := match[2]
+
+		if arg, exists := template.Data.Arguments[argName]; exists {
+			arg.Messages.Title = &flixkit.Title{I18N: map[string]string{langRE.FindStringSubmatch(codeCommentBlock)[1]: argTitle}}
+			template.Data.Arguments[argName] = arg
+		}
+	}
+
+	for _, match := range paramDescMatches {
+		argName := match[1]
+		argDesc := match[2]
+
+		if arg, exists := template.Data.Arguments[argName]; exists {
+			arg.Messages.Description = &flixkit.Description{I18N: map[string]string{langRE.FindStringSubmatch(codeCommentBlock)[1]: argDesc}}
+			template.Data.Arguments[argName] = arg
+		}
+	}
+
+	for _, match := range balanceMatches {
+		argName := match[1]
+		balance := match[2]
+
+		if arg, exists := template.Data.Arguments[argName]; exists {
+			arg.Balance = balance
+			template.Data.Arguments[argName] = arg
+		}
+	}
+
+	return nil
+}
+
+func regexpMatch(pattern, text string) (map[string]string, error) {
+	r := regexp.MustCompile(pattern)
+	names := r.SubexpNames()
+	match := r.FindStringSubmatch(text)
+	if match == nil {
+		return nil, nil
+	}
+
+	m := map[string]string{}
+	for i, n := range match {
+		m[names[i]] = n
+	}
+
+	return m, nil
+}
+
+func determineCadenceType(code string) (string, error) {
+	// Use regex to match only occurrences not in comments or strings.
+	transactionRegex := regexp.MustCompile(`(?s)\btransaction\s*(?:\([^)]*\))?\s*{.*`)
```

Comment:
the parsed cadence program can know if this is an transaction/script or interface. Using regex here is not the solution. Also with stable cadence these are wrong.